### PR TITLE
adding godoc synopsis for several top level packages

### PR DESCRIPTION
**Pull request by mikegray**
towards #2297 and #2298

### CODE REVIEW

File: go/chunks/chunk.go [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, version 2.0:
 // http://www.apache.org/licenses/LICENSE-2.0
 
+// Package chunks contains the core storage abstractions and implementations used in Noms.
```
Comment (aboodman) [inline go/chunks/chunk.go]: Trying to be a little more specific:

Package chunks provides facilities for representing, storing, and fetching content-addressed chunks of Noms data.

File: go/d/try.go [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, version 2.0:
 // http://www.apache.org/licenses/LICENSE-2.0
 
+// Package d implements several error and assertion functions used throughout Noms.
```
Comment (aboodman) [inline go/d/try.go]: Mention "debug", which is what "d" stands for.

Comment (mikegray) [inline go/d/try.go]: why don't we just name it `debug` then?

File: go/chunks/chunk.go (hunk 1 above)
Comment (mikegray) [inline go/chunks/chunk.go]: sounds good. done.

File: go/d/try.go (hunk 2 above)
Comment (cmasone-attic) [inline go/d/try.go]: `d` is shorter, essentially :-) Also, those of us from the Chromium world were used to using a macro called DCHECK() for stuff similar to this.

**Comment by cmasone-attic** [comment]
This seems fine to me, now, but Aaron had opinions, so I'd rather that he had the final sign off

**Comment by aboodman** [comment]
lgtm, thanks forthis

**Pull request merged**